### ISSUE TITLE
Error when installing from git+https://github.com/rethinkdb/rethinkdb.git#v1.12.0

### ISSUE THREAD

**Issue by HenryOrrin** [npm/npm#7812]
When installing directly form the git repository npm gives this error:
```
npm ERR! Cannot read property 'charAt' of undefined
npm ERR! 
```

npm-debug.log
```
0 info it worked if it ends with ok
1 verbose cli [ 'node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'install',
1 verbose cli   'git+https://github.com/rethinkdb/rethinkdb.git#v1.12.0' ]
2 info using npm@2.7.4
3 info using node@v0.12.1
4 verbose node symlink /usr/local/bin/node
5 silly cache add args [ 'git+https://github.com/rethinkdb/rethinkdb.git#v1.12.0',
5 silly cache add   null ]
6 verbose cache add spec git+https://github.com/rethinkdb/rethinkdb.git#v1.12.0
7 silly cache add parsed spec { raw: 'git+https://github.com/rethinkdb/rethinkdb.git#v1.12.0',
7 silly cache add   scope: null,
7 silly cache add   name: null,
7 silly cache add   rawSpec: 'git+https://github.com/rethinkdb/rethinkdb.git#v1.12.0',
7 silly cache add   spec: 'git+ssh://git@github.com/rethinkdb/rethinkdb.git#v1.12.0',
7 silly cache add   type: 'hosted',
7 silly cache add   hosted:
7 silly cache add    { type: 'github',
7 silly cache add      ssh: 'git@github.com:rethinkdb/rethinkdb.git#v1.12.0',
7 silly cache add      sshUrl: 'git+ssh://git@github.com/rethinkdb/rethinkdb.git#v1.12.0',
7 silly cache add      httpsUrl: 'https://github.com/rethinkdb/rethinkdb.git#v1.12.0',
7 silly cache add      directUrl: 'https://raw.githubusercontent.com/rethinkdb/rethinkdb/v1.12.0/package.json' } }
8 info maybeGithub Attempting git+https://github.com/rethinkdb/rethinkdb.git#v1.12.0 from git://github.com/rethinkdb/rethinkdb.git#v1.12.0
9 verbose addRemoteGit git-github-com-rethinkdb-rethinkdb-git-c3d0ff7b not in flight; caching
10 info git [ 'config', '--get', 'remote.origin.url' ]
11 verbose addRemoteGit remote.origin.url: git://github.com/rethinkdb/rethinkdb.git
12 verbose addRemoteGit updating existing cached remote /Users/vectorhacker/.npm/_git-remotes/git-github-com-rethinkdb-rethinkdb-git-c3d0ff7b
13 info git [ 'fetch', '-a', 'origin' ]
14 verbose addRemoteGit git fetch -a origin (git://github.com/rethinkdb/rethinkdb.git)
15 verbose addRemoteGit set permissions on /Users/vectorhacker/.npm/_git-remotes/git-github-com-rethinkdb-rethinkdb-git-c3d0ff7b
16 verbose addRemoteGit original treeish: v1.12.0
17 info git [ 'rev-list', '-n1', 'v1.12.0' ]
18 silly addRemoteGit resolved treeish: d48bc5031a7ce254031a37f76751f0247ed0fffd
19 verbose addRemoteGit resolved Git URL: git://github.com/rethinkdb/rethinkdb.git#d48bc5031a7ce254031a37f76751f0247ed0fffd
20 silly addRemoteGit Git working directory: /var/folders/68/f0cbz9gd2_3f4v42_py6pjf80000gn/T/npm-14488-3c218c46/git-cache-77aaa49cc4bf/d48bc5031a7ce254031a37f76751f0247ed0fffd
21 info git [ 'clone',
21 info git   '/Users/vectorhacker/.npm/_git-remotes/git-github-com-rethinkdb-rethinkdb-git-c3d0ff7b',
21 info git   '/var/folders/68/f0cbz9gd2_3f4v42_py6pjf80000gn/T/npm-14488-3c218c46/git-cache-77aaa49cc4bf/d48bc5031a7ce254031a37f76751f0247ed0fffd' ]
22 verbose addRemoteGit clone Cloning into '/var/folders/68/f0cbz9gd2_3f4v42_py6pjf80000gn/T/npm-14488-3c218c46/git-cache-77aaa49cc4bf/d48bc5031a7ce254031a37f76751f0247ed0fffd'...
22 verbose addRemoteGit done.
23 info git [ 'checkout', 'd48bc5031a7ce254031a37f76751f0247ed0fffd' ]
24 verbose addRemoteGit checkout Note: checking out 'd48bc5031a7ce254031a37f76751f0247ed0fffd'.
24 verbose addRemoteGit
24 verbose addRemoteGit You are in 'detached HEAD' state. You can look around, make experimental
24 verbose addRemoteGit changes and commit them, and you can discard any commits you make in this
24 verbose addRemoteGit state without impacting any branches by performing another checkout.
24 verbose addRemoteGit
24 verbose addRemoteGit If you want to create a new branch to retain commits you create, you may
24 verbose addRemoteGit do so (now or later) by using -b with the checkout command again. Example:
24 verbose addRemoteGit
24 verbose addRemoteGit   git checkout -b new_branch_name
24 verbose addRemoteGit
24 verbose addRemoteGit HEAD is now at d48bc50... Update release notes
25 error addLocal Could not install /var/folders/68/f0cbz9gd2_3f4v42_py6pjf80000gn/T/npm-14488-3c218c46/git-cache-77aaa49cc4bf/d48bc5031a7ce254031a37f76751f0247ed0fffd
26 info maybeGithub Couldn't clone git://github.com/rethinkdb/rethinkdb.git#v1.12.0
27 info maybeGithub Now attempting git+https://github.com/rethinkdb/rethinkdb.git#v1.12.0 from git@github.com:rethinkdb/rethinkdb.git#v1.12.0
28 verbose addRemoteGit git-github-com-rethinkdb-rethinkdb-git-ee2b972c not in flight; caching
29 info git [ 'config', '--get', 'remote.origin.url' ]
30 verbose addRemoteGit remote.origin.url: git@github.com:rethinkdb/rethinkdb.git
31 verbose addRemoteGit updating existing cached remote /Users/vectorhacker/.npm/_git-remotes/git-github-com-rethinkdb-rethinkdb-git-ee2b972c
32 info git [ 'fetch', '-a', 'origin' ]
33 verbose addRemoteGit git fetch -a origin (git@github.com:rethinkdb/rethinkdb.git)
34 verbose addRemoteGit set permissions on /Users/vectorhacker/.npm/_git-remotes/git-github-com-rethinkdb-rethinkdb-git-ee2b972c
35 verbose addRemoteGit original treeish: v1.12.0
36 info git [ 'rev-list', '-n1', 'v1.12.0' ]
37 silly addRemoteGit resolved treeish: d48bc5031a7ce254031a37f76751f0247ed0fffd
38 verbose stack TypeError: Cannot read property 'charAt' of undefined
38 verbose stack     at getResolved (/usr/local/lib/node_modules/npm/lib/cache/add-remote-git.js:357:13)
38 verbose stack     at /usr/local/lib/node_modules/npm/lib/cache/add-remote-git.js:220:23
38 verbose stack     at /usr/local/lib/node_modules/npm/lib/utils/git.js:26:8
38 verbose stack     at ChildProcess.exithandler (child_process.js:735:7)
38 verbose stack     at ChildProcess.emit (events.js:110:17)
38 verbose stack     at maybeClose (child_process.js:1008:16)
38 verbose stack     at Socket.<anonymous> (child_process.js:1176:11)
38 verbose stack     at Socket.emit (events.js:107:17)
38 verbose stack     at Pipe.close (net.js:476:12)
39 verbose cwd /Users/vectorhacker/Projects/parse-osmos-driver
40 error Darwin 14.1.0
41 error argv "node" "/usr/local/bin/npm" "install" "git+https://github.com/rethinkdb/rethinkdb.git#v1.12.0"
42 error node v0.12.1
43 error npm  v2.7.4
44 error Cannot read property 'charAt' of undefined
45 error If you need help, you may report this error at:
45 error     <https://github.com/npm/npm/issues>
46 verbose exit [ 1, true ]
```

**Comment by rmg** [comment]
Duplicate of #7807?

**Comment by HenryOrrin** [comment]
Indeed, haven't seen it before. Thanks.

**Comment by iarna** [comment]
The git related errors should all be fixed in `npm@2.8.0`. If you find you still have issues please open a new one.

**Comment by gdibble** [comment]
:rotating_light: https://github.com/npm/npm/issues/7807#issuecomment-97989999
